types/components/middleware/MiddlewareHandler.d.ts

```diff
@@ -0,0 +1,6 @@
+import { MiddlewareNext } from './MiddlewareNext';
+import { Request } from '../http/Request';
+import { Response } from '../http/Response';
+
+export type MiddlewarePromise = Promise<Error|void> | void;
+export type MiddlewareHandler = (request: Request, response: Response, next: MiddlewareNext) => MiddlewarePromise;
```

Comment:
I initially wrote this type wrong. The MiddlewarePromise should not ever be void but instead the MiddlewareHandler callback can accept a MiddlewarePromise or void. So just move the outside void from line 5 to 6 after the MiddlewarePromise.

Reply:
Fixed.